app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala

```
@@ -136,94 +119,102 @@ class SessionCache @Inject()(appConfig: AppConfig, mongo: ReactiveMongoComponent
+    putData(emailKey, Json.toJson(email)) map (_ => true)
 
-  def hasNino(implicit hc: HeaderCarrier): Future[Option[Boolean]] =
-    getCached[Option[Boolean]](sessionId, (cachedData, _) => cachedData.hasNino)
+  def saveHasNino(hasNino: Boolean)(implicit request: Request[_]): Future[Boolean] =
+    putData(hasNinoKey, Json.toJson(hasNino)) map (_ => true)
 
-  def mayBeEmail(implicit hc: HeaderCarrier): Future[Option[String]] =
-    getCached[Option[String]](sessionId, (cachedData, _) => cachedData.email)
+  def saveEori(eori: Eori)(implicit request: Request[_]): Future[Boolean] =
+    putData(eoriKey, Json.toJson(eori.id)) map (_ => true)
 
-  def safeId(implicit hc: HeaderCarrier): Future[SafeId] =
-    getCached[SafeId](sessionId, (cachedData, id) => cachedData.safeId(id))
+  def subscriptionDetails(implicit request: Request[_]): Future[SubscriptionDetails] =
+    getData[SubscriptionDetails](subDetailsKey).map(_.getOrElse(SubscriptionDetails()))
 
-  def name(implicit hc: HeaderCarrier): Future[Option[String]] =
-    getCached[Option[String]](sessionId, (cachedData, _) => cachedData.regDetails.map(_.name))
+  def eori(implicit request: Request[_]): Future[Option[String]] =
+    getData[String](eoriKey)
 
-  def registrationDetails(implicit hc: HeaderCarrier): Future[RegistrationDetails] =
-    getCached[RegistrationDetails](sessionId, (cachedData, id) => cachedData.registrationDetails(id))
+  def email(implicit request: Request[_]): Future[String] =
+    getData[String](emailKey).map(_.getOrElse(throwException(emailKey)))
 
-  def registerWithEoriAndIdResponse(implicit hc: HeaderCarrier): Future[RegisterWithEoriAndIdResponse] =
-    getCached[RegisterWithEoriAndIdResponse](
-      sessionId,
-      (cachedData, id) => cachedData.registerWithEoriAndIdResponse(id)
-    )
+  def hasNino(implicit request: Request[_]): Future[Option[Boolean]] =
+    getData[Boolean](hasNinoKey) // .map(_.getOrElse(throwException(hasNinoKey)))
```

Is it intended behaviour that we dont map this value here?

---

Yes. the getData call returns an Option. Sometime we want to return an Option to the client. Other times we want to throw an exception when we always expect a value. I'll delete the commented out code.